### PR TITLE
Replace macro __HIP_PLATFORM_HCC__ with __HIP_PLATFORM_AMD__

### PR DESCRIPTION
**Pull request by wenkaidu**

### CODE REVIEW

File: src/p2p_plugin.c [hunk 1]
```diff
@@ -106,7 +106,7 @@ ncclResult_t nccl_p2p_gdr_support(int dev)
   static int module_loaded = -1;
 
   if (module_loaded == -1) {
-#if defined(__HIP_PLATFORM_HCC__) || defined(__HCC__) || defined(__HIPCC__)
+#if defined(__HIP_PLATFORM_AMD__) || defined(__HIPCC__)
```
Comment (nusislam) [inline src/p2p_plugin.c:109]: For my understanding, why is this change required?

Comment (wenkaidu) [reply src/p2p_plugin.c:109]: SWDEV-427469
